pkg/docker/config/config.go

```diff
@@ -245,6 +326,48 @@ func RemoveAllAuthentication(sys *types.SystemContext) error {
 	})
 }
 
+func removeAuthFromCredConfigs(credConfigs []string, registry string) error {
+	var err error
+	for _, credConfig := range credConfigs {
+		if _, err = getAuthFromCredHelper(credConfig, registry); err != nil {
+			if credentials.IsErrCredentialsNotFoundMessage(err.Error()) {
+				logrus.Warnf("Not logged in to %s with credential helper %s", registry, credConfig)
+				err = ErrNotLoggedIn
+			}
+			break
+		}
+		if err = deleteAuthFromCredHelper(credConfig, registry); err != nil {
+			return err
+		}
+		logrus.Debugf("Credentials were deleted from credential helper %s", credConfig)
+		return nil
+	}
+	return err
+}
+
+func removeAllAuthFromCredConfigs(credConfigs []string) error {
+	var err error
+	for _, credHelper := range credConfigs {
+		creds, err := listAuthFromCredHelper(credHelper)
+		if err != nil {
+			return err
+		}
+		for registry := range creds {
+			if err = deleteAuthFromCredHelper(credHelper, registry); err != nil {
+				logrus.Warningf("error deleting credentials for %s from %s: %v", registry, credHelper, err)
+			}
+		}
+	}
+
+	return err
```

Comment:
This currently fairly randomly returns an error or `nil`, depending on what happened last. That doesn’t make much sense.

I’m not quite sure how to handle this _well_, though — listing every single credential that failed might be dozens or hundreds in pathological cases. But it’s probably a reasonable starting point — let’s not lose data, and we can maybe somehow summarize later if it turns out to be necessary.